### PR TITLE
PMM-8865 added new test for case 207 to verify hovering over query in overview table

### PR DESCRIPTION
**Pull request by ElenaFedichkina**
Added new test for verification of Query on QAN overview table
https://pmm.cd.percona.com/job/pmm2-ui-test-nightly/853/
image
https://pmm.cd.percona.com/job/pmm2-ui-test-nightly/860/

### CODE REVIEW

File: tests/QAN/overview_test.js [hunk 1]
```diff
@@ -9,6 +11,28 @@ Before(async ({
   qanFilters.waitForFiltersToLoad();
 });
 
+Scenario(
+  'PMM-T207 Verify hovering over query in overview table  @qan',
+  async ({ I, qanOverview }) => {
+    qanOverview.waitForOverviewLoaded();
+    const firstQueryLocator = `${qanOverview.getRowLocator(1)} div.td:nth-child(2) div`;
```
Comment (puneet0191) [inline tests/QAN/overview_test.js:18]: no other locator possible?

File: tests/QAN/overview_test.js [hunk 2]
```diff
@@ -9,6 +11,28 @@ Before(async ({
...
+
+    firstQueryText = qanOverview.removeSymbolFromString(firstQueryText, ' ');
+
+    I.moveCursorTo(`${firstQueryLocator} > svg`);
+    I.waitForVisible(qanOverview.elements.tooltipQueryValue, 30);
+    I.seeElement(qanOverview.elements.tooltipQueryValue);
+    let tooltipQueryText = await I.grabTextFrom(qanOverview.elements.tooltipQueryValue);
+
+    tooltipQueryText = qanOverview.removeSymbolFromString(tooltipQueryText, ' ');
+    tooltipQueryText = qanOverview.removeSymbolFromString(tooltipQueryText, '\n');
```
Comment (puneet0191) [inline tests/QAN/overview_test.js:31]: you could do the same in one line

Comment (puneet0191) [reply tests/QAN/overview_test.js:31]: did you try string.replace function to achieve the same result?

**Review by puneet0191 — APPROVED** [review]
just some minor comments, see if you can address them. Good stuff

File: tests/QAN/overview_test.js (hunk 1 above)
Comment (ElenaFedichkina) [reply tests/QAN/overview_test.js:18]: no. i think it can't be better(

File: tests/QAN/overview_test.js (hunk 2 above)
Comment (ElenaFedichkina) [reply tests/QAN/overview_test.js]: fixed. thanks

File: tests/QAN/overview_test.js [hunk 3]
```diff
@@ -9,6 +11,27 @@ Before(async ({
...
+
+    firstQueryText = firstQueryText.replace(/ /g, '');
+
+    I.moveCursorTo(`${firstQueryLocator} > svg`);
+    I.waitForVisible(qanOverview.elements.tooltipQueryValue, 30);
+    I.seeElement(qanOverview.elements.tooltipQueryValue);
+    let tooltipQueryText = await I.grabTextFrom(qanOverview.elements.tooltipQueryValue);
+
+    tooltipQueryText = tooltipQueryText.replace(/ /g, '').replace(/\n/g, '');
+    assert.ok(firstQueryText === tooltipQueryText, 'Expected that the request text in the line matches the request text on the tooltip');
```
Comment (puneet0191) [inline tests/QAN/overview_test.js:31]: please improve the assert message to include both expect and actual text, so when the test fails it becomes easy to understand why it does

File: tests/QAN/overview_test.js [hunk 4]
```diff
@@ -9,6 +11,27 @@ Before(async ({
   qanFilters.waitForFiltersToLoad();
 });
 
+Scenario(
+  'PMM-T207 Verify hovering over query in overview table  @qan',
+  async ({ I, qanOverview }) => {
+    qanOverview.waitForOverviewLoaded();
+    const firstQueryLocator = `${qanOverview.getRowLocator(1)} div.td:nth-child(2) div`;
```
Comment (puneet0191) [inline tests/QAN/overview_test.js:18]: do you have any specific reason for including all these inside the scenario file? why can't locators and logic go to overview_fragment, thats how it works for all other scenarios too.

**Review by puneet0191 — CHANGES_REQUESTED** [review]

File: tests/QAN/overview_test.js (hunk 3 above)
Comment (ElenaFedichkina) [reply tests/QAN/overview_test.js:31]: ok

**Review by puneet0191 — APPROVED** [review]

**Pull request merged**